aten/src/ATen/native/xnnpack/OpContext.cpp

```diff
@@ -0,0 +1,59 @@
+#ifdef USE_XNNPACK
+#include <ATen/native/xnnpack/Convolution.h>
+#include <ATen/native/xnnpack/Linear.h>
+#include <ATen/native/xnnpack/OpContext.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+
+c10::intrusive_ptr<XNNPackLinearOpContext>
+XNNPackLinearOpContext::create_context(at::Tensor&& weight,
+    c10::optional<at::Tensor>&& bias,
```

Comment:
rvalue references cannot bind to lvalues by design.  The accepted paradigm is to either (1) pass parameters by value, (2) pass by [const] lvalue reference if copies are expensive, or (3) provide a (const lvalue reference, rvalue reference) pair overload.  Providing an rvalue reference only interface limits use.

By the way, tensors in PyTorch are ref-counted wrappers to internal implementations so they are cheap to shallow-copy around.

Reply:
> Providing an rvalue reference only interface limits use.

So this was by intention. I wanted to make it explicit that callsite has to provide rvalue reference. Doing by value introduces shallow copy that are ref counted, and ref counting is not exactly cheap, right? So I thought this made more sense. 
But I might be mistaken, so do provide more context if you disagree and I can learn something new.
We can also do options 3 that you suggested, but I wanted to make sure I provide the context for my reasoning.

Reply:
Agreed, shallow ref-counted copy is not cheap.

Reply:
I think the question is whether you envision a case where create_context is called with weight and bias tensors that are neither temporaries nor moved-from objects?  In other words, do we want to be able to  support something like this:

```
Tensor weight;  // Some weight tensor somewhere
create_context( std::move(weight) );
weight.blah();  // weight is moved from.  It can no longer be used so this is not valid.
```

Reply:
Yes, so I did not envision a former usecase. In my mind weight is always moved from while creating context and thus not valid after that.

Reply:
Sounds good, thanks!